### PR TITLE
Automatically add the PR link to the Asana task

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1199333091098016/1204360503924262/f
iOS PR: N/A
macOS PR: N/A
What kind of version bump will this require?: No release required, this is a workflow change only

**Optional**:

CC: @ayoy 

**Description**:

This PR adds a GitHub Actions workflow to automatically add a PR link to the referenced Asana task.

**Steps to test this PR**:
1. Quick approach: check the Asana task and verify that a comment was posted
1. More in depth approach: create a branch off of this branch, make a public Asana task, then open a PR which references it and validate that the comment appears

Also, we want to test that draft PRs don't trigger this workflow. You can see [this PR](https://github.com/duckduckgo/BrowserServicesKit/pull/386) which includes a comment to the skipped CI run, or you can create a branch from this PR and open a new draft PR with it to be sure.

**OS Testing**:

* N/A

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by ayoy** [comment]
Thanks for adding this @samsymons! I have just one remark here, which is that this workflow will be getting in the way of the recently rolled out BSK PR automation. A comment is being posted to Asana for that automation already (with links to 3 PRs). I'm fine with adjusting the workflow to not post the comment, but we need to:
* roll out this automation in all 3 repos: BSK, iOS and macOS,
* accept that we're fine with 3 Dax comments to Asana task for a BSK change.
Have you considered this?

**Review by ayoy — APPROVED** [review]
This of course works great (as tested [here](https://github.com/duckduckgo/BrowserServicesKit/pull/313)) :) Let's wait for the outcome of the discussion on this proposal but code-wise this change is flawless 💪

**Comment by samsymons** [comment]
Testing the `reopened` event to make sure that the automation fires again - though, tbh I feel like that event should be removed before merging this, unless there's a good cause to keep it.

File: .github/workflows/add-asana-pr-comment.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Add PR link to Asana task
+
+on:
+  pull_request:
+    types: [opened, reopened]
```
Comment (samsymons) [inline .github/workflows/add-asana-pr-comment.yml:5]: Tempted to change this to the following, since I can't think of a good reason that we would need to trigger this on `reopened`:

```suggestion
    types: [opened]
```

**Comment by samsymons** [comment]
@ayoy I've updated this PR to ignore draft PRs, as we discussed today. I tested this in a draft PR which is linked to in the task description.

File: .github/workflows/add-asana-pr-comment.yml (hunk 1 above)
Comment (ayoy) [reply .github/workflows/add-asana-pr-comment.yml:5]: Very good point @samsymons, let's limit it only to opened event. I'll try to commit the suggestion :)

File: .github/workflows/add-asana-pr-comment.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Add PR link to Asana task
+
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  sync:
```
Comment (ayoy) [inline .github/workflows/add-asana-pr-comment.yml:8]: I'd just rename this step, because it seems unrelated to the actual action being performed :)

```suggestion
  add-asana-comment:
    name: Add Asana comment
```

**Review by ayoy — APPROVED** [review]
That looks great, thanks @samsymons! I've checked the runs you've linked in your comment above and also the workflow code is straightforward so it's good to go for me. Let's just update the job name :)

File: .github/workflows/add-asana-pr-comment.yml (hunk 2 above)
Comment (samsymons) [reply .github/workflows/add-asana-pr-comment.yml:8]: Good catch, thanks!

**Comment by samsymons** [comment]
@shakyShane @GioSensation I have a change here which automatically posts PR links to Asana tasks, is this going to affect any of your workflows on the frontend side?
